### PR TITLE
[Windows] Add support deploy Windows ovf template and collect VM settings info

### PR DESCRIPTION
**Pull request by Tomorrow9**
1. add support deploy Windows ovf template as Linux,
2. collect deployed VM hardware configurations info

### CODE REVIEW

**Comment by Tomorrow9** [comment]
```
VM information:
+------------------------------------------------------------------------------------------+
| Name                      | test_vm_1654867469758                                        |
+------------------------------------------------------------------------------------------+
| IP                        |                                                                        |
+------------------------------------------------------------------------------------------+
| Guest OS Distribution     |                                                              |
+------------------------------------------------------------------------------------------+
| Hardware Version          | vmx-19                                                       |
+------------------------------------------------------------------------------------------+
| VMTools Version           | 11.3.5 build 18557794                                        |
+------------------------------------------------------------------------------------------+
| Config Guest Id           | windows9_64Guest                                             |
+------------------------------------------------------------------------------------------+
| GuestInfo Guest Id        | windows9_64Guest                                             |
+------------------------------------------------------------------------------------------+
| GuestInfo Guest Full Name | Microsoft Windows 10 (64-bit)                                |
+------------------------------------------------------------------------------------------+
| GuestInfo Guest Family    | windowsGuest                                                 |
+------------------------------------------------------------------------------------------+
| GuestInfo Detailed Data   | architecture='X86'                                           |
|                           | bitness='64'                                                 |
|                           | buildNumber='22000'                                          |
|                           | distroName='Windows'                                         |
|                           | distroVersion='10.0'                                         |
|                           | familyName='Windows'                                         |
|                           | kernelVersion='22000.593'                                    |
|                           | prettyName='Windows 10 Enterprise, 64-bit (Build 22000.593)' |
+------------------------------------------------------------------------------------------+


Test Results (Total: 1, Passed: 1, Elapsed Time: 00:16:01)
+------------------------------------+
| Name          | Status | Exec Time |
+------------------------------------+
| deploy_vm_ovf | Passed | 00:15:44  |
+------------------------------------+
```

File: windows/deploy_vm/collect_ovf_vm_config.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  when:
+    - item._vimtype in ['vim.vm.device.VirtualVmxnet3', 'vim.vm.device.VirtualE1000e', 'vim.vm.device.VirtualE1000']
+  with_items: "{{ vm_config.config.hardware.device }}"
+
+- name: Set fact of vTPM device existence info
+  set_fact:
+    ovf_vm_vtpm_info: True
+  when:
+    - item._vimtype == "vim.vm.device.VirtualTPM"
+  with_items: "{{ vm_config.config.hardware.device }}"
```
Comment (keirazhang) [inline windows/deploy_vm/collect_ovf_vm_config.yml:68]: I think here can be changed to:
```
set_fact:
  ovf_vm_vtpm_info:  "{{ vm_config.config.hardware.device | selectattr('_vimtype', 'equalto', 'vim.vm.device.VirtualTPM') | length > 0 }}"
```

Comment (Tomorrow9) [reply windows/deploy_vm/collect_ovf_vm_config.yml]: Yeah, changed it. thanks.

**Comment by Tomorrow9** [comment]
Windows ovf template:
```
VM information:
+------------------------------------------------------------------------------------------+
| Name                      | test_vm                                                      |
+------------------------------------------------------------------------------------------+
| IP                        |                                           |
+------------------------------------------------------------------------------------------+
| Guest OS Distribution     |                                                              |
+------------------------------------------------------------------------------------------+
| Hardware Version          | vmx-19                                                       |
+------------------------------------------------------------------------------------------+
| VMTools Version           | 11.3.5 (build-18557794)                                      |
+------------------------------------------------------------------------------------------+
| Config Guest Id           | windows9_64Guest                                             |
+------------------------------------------------------------------------------------------+
| GuestInfo Guest Id        | windows9_64Guest                                             |
+------------------------------------------------------------------------------------------+
| GuestInfo Guest Full Name | Microsoft Windows 10 (64-bit)                                |
+------------------------------------------------------------------------------------------+
| GuestInfo Guest Family    | windowsGuest                                                 |
+------------------------------------------------------------------------------------------+
| GuestInfo Detailed Data   | architecture='X86'                                           |
|                           | bitness='64'                                                 |
|                           | buildNumber='22000'                                          |
|                           | distroName='Windows'                                         |
|                           | distroVersion='10.0'                                         |
|                           | familyName='Windows'                                         |
|                           | kernelVersion='22000.593'                                    |
|                           | prettyName='Windows 10 Enterprise, 64-bit (Build 22000.593)' |
+------------------------------------------------------------------------------------------+


Test Results (Total: 1, Passed: 1, Elapsed Time: 01:30:24)
+------------------------------------+
| Name          | Status | Exec Time |
+------------------------------------+
| deploy_vm_ovf | Passed | 00:56:52  |
+------------------------------------+
```
Photon OS ova template:
VM information:
+-----------------------------------------------------------------+
| Name                      | test_photon4_ova                    |
+-----------------------------------------------------------------+
| IP                        |                      |
+-----------------------------------------------------------------+
| Guest OS Distribution     | VMware Photon OS 4.0 x86_64         |
+-----------------------------------------------------------------+
| Hardware Version          |                              |
+-----------------------------------------------------------------+
| VMTools Version           | 12.x.x (build-19953104)             |
+-----------------------------------------------------------------+
| CloudInit Version         | 21.4                                |
+-----------------------------------------------------------------+
| GUI Installed             | False                               |
+-----------------------------------------------------------------+
| Config Guest Id           | vmwarePhoton64Guest                 |
+-----------------------------------------------------------------+
| GuestInfo Guest Id        | vmwarePhoton64Guest                 |
+-----------------------------------------------------------------+
| GuestInfo Guest Full Name | VMware Photon OS (64-bit)           |
+-----------------------------------------------------------------+
| GuestInfo Guest Family    | linuxGuest                          |
+-----------------------------------------------------------------+
| GuestInfo Detailed Data   | architecture='X86'                  |
|                           | bitness='64'                        |
|                           | distroName='VMware Photon OS'       |
|                           | distroVersion='4.0'                 |
|                           | familyName='Linux'                  |
|                           | kernelVersion='5.10.83-6.ph4-esx'   |
|                           | prettyName='VMware Photon OS/Linux' |
+-----------------------------------------------------------------+


Test Results (Total: 6, Passed: 6, Elapsed Time: 00:22:46)
+-------------------------------------------+
| Name                 | Status | Exec Time |
+-------------------------------------------+
| check_inbox_driver   | Passed | 00:07:03  |
| ovt_verify_install   | Passed | 00:11:10  |
| ovt_verify_status    | Passed | 00:01:09  |
| vgauth_check_service | Passed | 00:01:07  |
| check_ip_address     | Passed | 00:00:57  |
| check_os_fullname    | Passed | 00:00:57  |
+-------------------------------------------+
```

**Review by keirazhang — APPROVED** [review]
LGTM. Thanks~

**Pull request merged**